spec/views/hq/sponsors/_form_spec.rb

```diff
@@ -0,0 +1,165 @@
+    assign :sponsor_types, SponsorType.all
+    assign :organizations, Organization.all
+    assign :branches, Branch.all
+    assign :cities, [sponsor_random_optional_fields.city].unshift(Sponsor::NEW_CITY_MENU_OPTION)
+  end
+
+  specify 'has a form' do
+    render
+
+    assert_select 'form'
+  end
+
+  describe 'has a "Cancel" button' do
+    specify 'using an existing Sponsor record' do
+      allow(sponsor_random_optional_fields).to receive(:id).and_return sponsor_random_optional_fields.id
+      render
+
+      assert_select 'a[href=?]', hq_sponsor_path(sponsor_random_optional_fields.id), text: 'Cancel'
+    end
+
+    specify 'using a new Sponsor record' do
+      allow(sponsor_random_optional_fields).to receive(:id).and_return nil
+      render
+
+      assert_select 'a[href=?]', hq_sponsors_path, text: 'Cancel'
+    end
+  end
+
+  describe 'has form values' do
+    specify 'using an existing Sponsor record' do
+      allow(User).to receive(:pluck).and_return([user.user_name, user.id])
+      render
+
+      assert_select "input#sponsor_name" do
+        assert_select "[value=?]", CGI::escape_html(sponsor_random_optional_fields.name)
+      end
+
+      assert_select "select#sponsor_status_id" do
+        assert_select "option", value: sponsor_random_optional_fields.status_id,
+                                html: CGI::escape_html(sponsor_random_optional_fields.status.name)
+      end
+
+      assert_select "select#sponsor_gender" do
+        assert_select "option", value: Settings.lookup.gender.first,
+                                html: CGI::escape_html(Settings.lookup.gender.first)
+      end
+
+      assert_select "input#sponsor_start_date" do
+        assert_select "[value=?]", sponsor_random_optional_fields.start_date
+      end
+
+      assert_select "input#sponsor_requested_orphan_count" do
+        assert_select "[value=?]", sponsor_random_optional_fields.requested_orphan_count
+      end
+
+      assert_select "input#sponsor_request_fulfilled" do
+        assert_select "[disabled=?]", "disabled"
+        assert_select "[value=?]", sponsor_random_optional_fields.request_fulfilled ? "1" : "0"
+        assert_select "[checked]", true if sponsor_random_optional_fields.request_fulfilled
+      end
+
+      assert_select "select#sponsor_sponsor_type_id" do
+        assert_select "[disabled=?]", "disabled"
+        assert_select "option", value: sponsor_random_optional_fields.sponsor_type_id,
+                                html: CGI::escape_html(sponsor_random_optional_fields.sponsor_type.name) do
+            assert_select "[selected=?]", "selected"
+        end
+      end
+
+      assert_select "select#sponsor_organization_id" do
+        assert_select "[disabled=?]", "disabled"
+        assert_select "option", value: sponsor_random_optional_fields.organization_id
+        assert_select "option", html: CGI::escape_html(sponsor_random_optional_fields.organization.name) if sponsor_random_optional_fields.organization
```

Comment:
we tend to prefer an 80 char line width and only use a trailing if, when it can fit into these constraints.
perhaps a smaller variable name such as `full_sponsor` or `sponsor_with_opts`.